src/shared/serialize.c

```diff
@@ -403,6 +406,44 @@ int deserialize_pidref(FDSet *fds, const char *value, PidRef *ret) {
         return 0;
 }
 
+int deserialize_fd_from_set(const char *value, FDSet *fds) {
+        int fd, r;
+
+        assert(value);
+        assert(fds);
+
+        fd = parse_fd(value);
+        if (fd < 0)
+                return log_debug_errno(fd, "Failed to parse FD out of value: %s", value);
+
+        if (!fdset_contains(fds, fd))
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "FD %d not in fdset.", fd);
```

Comment:
this check is redundant, fdset_remove() cecks for that too.

also, can we please get #29481 merged, it adds a deserializer for this, and moves everything over.

Reply:
it's dropped